### PR TITLE
feat: Update Mistral version and support tool calling

### LINKED ISSUE #819: [Feature Request] Migrate Mistral AI to 1.0.0
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

Migrate Mistral AI to 1.0.0

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

Update Mistral version and support tool calling

## Motivation and Context

`close #819 ` 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/models/role_playing_with_mistral.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    SEARCH_FUNCS,
+)
+from camel.types import ModelPlatformType, ModelType
+from camel.utils import print_text_animated
+
+
+def main(
+    model_platform=ModelPlatformType.MISTRAL,
+    model_type=ModelType.MISTRAL_LARGE,
+    chat_turn_limit=10,
```
Comment (raywhoelse) [inline examples/models/role_playing_with_mistral.py:34]: IMO, Is it better to move it into a function?

File: camel/models/mistral_model.py [hunk 2]
```diff
@@ -15,24 +15,35 @@
...
 
+try:
+    import os
+
+    if os.getenv("AGENTOPS_API_KEY") is not None:
+        from agentops import LLMEvent, record
+    else:
+        raise ImportError
+except (ImportError, AttributeError):
+    LLMEvent = None
```
[the remark this answers is not in the document]
Comment (Wendong-Fan) [reply camel/models/mistral_model.py:41]: AgentOps is not required, if user want to use this to track agent running, they will set AgentOps key, then we will run code for AgentOps

File: examples/models/role_playing_with_mistral.py (hunk 1 above)
Comment (Wendong-Fan) [reply examples/models/role_playing_with_mistral.py:34]: This is example code we don't need to add more abstraction for example code

File: camel/configs/mistral_config.py [hunk 3]
```diff
@@ -53,10 +56,11 @@ class MistralConfig(BaseConfig):
...
-    max_tokens: Optional[int] = None
     top_p: Optional[float] = None
+    max_tokens: Optional[int] = None
+    min_tokens: Optional[int] = None
+    stop: Optional[Union[str, list[str]]] = None
     random_seed: Optional[int] = None
-    safe_mode: bool = False
     safe_prompt: bool = False
     response_format: Optional[Union[Dict[str, str], Any]] = None
     tool_choice: Optional[str] = "auto"
```
[the remark this answers is not in the document]
Comment (Wendong-Fan) [reply camel/configs/mistral_config.py:66]: it has been moved into camel/camel/configs/base_config.py

**Pull request merged**